packages/hashi/src/mainClient.js

```diff
@@ -104,6 +120,94 @@ export default class MainClient {
       });
     });
   }
+
+  __fetchContentData(message) {
+    // based on the incoming information, get data
+    // from Kolibri to pass back to the iframe
+
+    // if filtering by optional params
+    if (message.options) {
+      let getParams = {};
+      let options = message.options;
+      if (options.parent && options.parent == 'self') {
+        // need to fetch this value when this
+        // function is move to a location that has access
+        // to `content`
+        // getParams.parent = rootNode;
+      } else if (options.parent) {
+        getParams.parent = options.parent;
+      }
+      options.ids ? (getParams.ids = options.ids) : null;
+      options.page ? (getParams.page = options.page) : null;
+      options.pageSize ? (getParams.ids = options.pageSize) : null;
+      ContentNodeResource.fetchCollection({ getParams }).then(contentNodes => {
+        contentNodes ? (message.status = 'success') : (message.status = 'failure');
+        let response = {};
+        response.page = message.options.page ? message.options.page : 1;
+        response.pageSize = message.options.pageSize ? message.options.pageSize : 50;
+        response.results = contentNodes;
+        message.data = response;
+        message.type = 'response';
+        this.mediator.sendMessage({
+          nameSpace,
+          event: events.DATARETURNED,
+          data: message,
+        });
+      });
+    }
+    // or, if getting by a specific id
+    else if (message.id) {
+      let id = message.id;
+      ContentNodeResource.fetchModel({ id }).then(contentNode => {
+        if (contentNode) {
+          message.status = 'success';
+        } else {
+          message.status = 'failure';
+        }
+        message.data = contentNode;
+        message.type = 'response';
+        this.mediator.sendMessage({
+          nameSpace,
+          event: events.DATARETURNED,
+          data: message,
+        });
+      });
+    }
+  }
+
+  __navigateTo(message) {
+    let id = message.nodeId;
+    ContentNodeResource.fetchModel({ id }).then(contentNode => {
+      let routeBase, context;
+      if (contentNode && contentNode.kind === 'topic') {
+        routeBase = '/topics/t';
+      } else if (contentNode) {
+        routeBase = '/topics/c';
+      }
+      if (!message.context) {
+        // if there is custom context, don't re-route
+        const path = `${routeBase}/${id}`;
+        router.push({ path: path, query: { context: context } }).catch(() => {});
```

Comment:
To save implementation work by all of the custom channels apps - we can probably update the `node_id` property of the context ourselves here, in the case when we are showing the render overlay over the custom channel navigation.